screens/Scan/ScanScreenController.ts

```diff
@@ -66,6 +71,16 @@ export function useScanScreen({ navigation }: MainRouteProps) {
     };
   }, []);
 
+  useEffect(() => {
+    SystemSetting.isAirplaneEnabled().then((enable) => {
+      if (enable) {
+        scanService.send(ScanEvents.FLIGHT_ENABLED());
+      } else {
+        scanService.send(ScanEvents.FLIGHT_DISABLED());
+      }
+    });
+  }, [isActive]);
```

Comment:
Have you tried using the `APP_ACTIVE` event in the state machine? It fires from the parent app machine when the focus changes.